relay-server/src/processing/logs/integrations/vercel.rs

```
@@ -0,0 +1,52 @@
+use relay_event_schema::protocol::OurLog;
+use relay_ourlogs::VercelLog;
+
+use crate::integrations::VercelLogDrainFormat;
+use crate::processing::logs::{Error, Result};
+use crate::services::outcome::DiscardReason;
+
+/// Expands Vercel logs into the [`OurLog`] format.
+pub fn expand<F>(format: VercelLogDrainFormat, payload: &[u8], mut produce: F) -> Result<()>
+where
+    F: FnMut(OurLog),
+{
+    let mut count: i32 = 0;
+
+    match format {
+        VercelLogDrainFormat::Json => {
+            let logs = serde_json::from_slice::<Vec<VercelLog>>(payload).map_err(|e| {
+                relay_log::debug!(
+                    error = &e as &dyn std::error::Error,
+                    "Failed to parse logs data as JSON"
+                );
+                Error::Invalid(DiscardReason::InvalidJson)
+            })?;
+
+            for log in logs {
+                count += 1;
+                let ourlog = relay_ourlogs::vercel_log_to_sentry_log(log);
+                produce(ourlog)
+            }
+        }
+        VercelLogDrainFormat::NDJson => {
+            for line in payload.split(|&b| b == b'\n') {
+                if line.is_empty() {
+                    continue;
+                }
+
+                if let Ok(log) = serde_json::from_slice::<VercelLog>(line) {
```

I think we wouldn't want to swallow errors here and instead emit a single error.

This would mean you need to pass in the `RecordKeeper` and emit an error. But we can also tackle this separately/afterwards.

We should then also make a test which has a single invalid line.

Then this maybe also addresses the issue with the `count` afterwards.

---

I agree. I would like to come back to this in another PR and refactor this to improve the error handling + handle counts.

For now can we merge this in to unblock? I will keep iterating.

---

Yep of course 👍